### PR TITLE
Fix PyTorch versions in Lite CI

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

Lite's "oldest" CI wasn't correctly installing 1.9 (discovered by @otaj)

This PR fixes it.

The list of jobs now matches PyTorch's CI

### Does your PR introduce any breaking changes? If yes, please list them.

None

cc @carmocca @akihironitta @borda @justusschock @awaelchli

### CODE REVIEW

File: .github/workflows/ci-lite-tests.yml [hunk 1]
```diff
@@ -101,11 +100,14 @@ jobs:
         restore-keys: |
           ${{ runner.os }}-pip-py${{ matrix.python-version }}-${{ matrix.pkg-name }}-${{ matrix.release }}-${{ matrix.requires }}-
 
-    - name: Install dependencies
+    - name: Switch PyTorch URL
+      run: python -c "print('TORCH_URL=https://download.pytorch.org/whl/' + str('test/cpu/torch_test.html' if '${{matrix.release}}' == 'pre' else 'cpu/torch_stable.html'))" >> $GITHUB_ENV
+
+    - name: Install package & dependencies
```
Comment (carmocca) [inline .github/workflows/ci-lite-tests.yml:106]: Some of these changes come from https://github.com/Lightning-AI/lightning/pull/15105

File: .github/workflows/ci-lite-tests.yml [hunk 2]
```diff
@@ -33,19 +33,27 @@ jobs:
...
-        release: ["stable"]
-        exclude:
-          # note: there's no distribution of Torch==1.9 for Python>=3.9
-          # TODO: Remove the exclusion when dropping PyTorch 1.9 support.
-          - {python-version: "3.9", requires: "oldest"}
-          - {python-version: "3.10", requires: "oldest"}
         include:
-          #- {os: ubuntu-22.04, pkg-name: "LAI", python-version: "3.10", release: "pre"}
-          - {os: ubuntu-22.04, pkg-name: "lite", python-version: "3.10", release: "pre"}
+          # assign python and pytorch version combinations to operating systems (arbitrarily)
```
Comment (carmocca) [inline .github/workflows/ci-lite-tests.yml:37]: This list now matches the one for `pl-cpu` jobs

File: .github/workflows/ci-lite-tests.yml [hunk 3]
```diff
@@ -72,20 +80,11 @@ jobs:
       run: |
         python .actions/assistant.py replace_oldest_ver
 
-    - name: switch PyTorch URL
-      run: python -c "print('TORCH_URL=https://download.pytorch.org/whl/' + str('test/cpu/torch_test.html' if '${{matrix.release}}' == 'pre' else 'cpu/torch_stable.html'))" >> $GITHUB_ENV
-
-    - name: Install package
-      env:
-        PACKAGE_NAME: ${{ matrix.pkg-name }}
+    - name: Adjust PyTorch versions in requirements files
```
Comment (carmocca) [inline .github/workflows/ci-lite-tests.yml:83]: I believe this step was missing

Comment (otaj) [reply .github/workflows/ci-lite-tests.yml:83]: So, it seems it's a combination of these two issues:
This step was missing, but also changes made in #15337 were missing too

Comment (carmocca) [reply .github/workflows/ci-lite-tests.yml:83]: Teamwork!

**Review by justusschock — APPROVED** [review]

**Review by awaelchli — APPROVED** [review]
All I see is yaml

File: .github/checkgroup.yml [hunk 4]
```diff
@@ -150,24 +150,21 @@ subprojects:
...
-      - "lite-cpu (ubuntu-20.04, lite, 3.8, oldest, stable)"
-      - "lite-cpu (ubuntu-20.04, lite, 3.9, latest, stable)"
-      - "lite-cpu (ubuntu-20.04, lite, 3.10, latest, stable)"
-      - "lite-cpu (windows-2022, lite, 3.7, latest, stable)"
-      - "lite-cpu (windows-2022, lite, 3.7, oldest, stable)"
-      - "lite-cpu (windows-2022, lite, 3.8, latest, stable)"
-      - "lite-cpu (windows-2022, lite, 3.8, oldest, stable)"
-      - "lite-cpu (windows-2022, lite, 3.9, latest, stable)"
-      - "lite-cpu (windows-2022, lite, 3.10, latest, stable)"
+      - "lite-cpu (macOS-11, lite, 3.9, 1.11)"
```
Comment (Borda) [inline .github/checkgroup.yml:153]: can we keep the sorted by OS so it is easier to read...

Comment (Borda) [reply .github/checkgroup.yml:153]: did the re-order, feel free to rever... @carmocca

**Review by Borda — APPROVED** [review]

**Pull request merged**